packages/cli/src/controllers/dynamicNodeParameters.controller.ts

```diff
@@ -0,0 +1,153 @@
+import { Service } from 'typedi';
+import type { RequestHandler } from 'express';
+import { NextFunction, Response } from 'express';
+import type {
+	INodeCredentials,
+	INodeListSearchResult,
+	INodeParameters,
+	INodePropertyOptions,
+	INodeTypeNameVersion,
+	ResourceMapperFields,
+} from 'n8n-workflow';
+import { jsonParse } from 'n8n-workflow';
+
+import { Authorized, Get, Middleware, RestController } from '@/decorators';
+import { getBase } from '@/WorkflowExecuteAdditionalData';
+import { DynamicNodeParametersService } from '@/services/dynamicNodeParameters.service';
+import { BadRequestError } from '@/ResponseHelper';
+import type { AuthenticatedRequest } from '@/requests';
+
+const assertMethodName: RequestHandler = (req, res, next) => {
+	const { methodName } = req.query as BaseRequest['query'];
+	if (!methodName) {
+		throw new BadRequestError('Parameter methodName is required.');
+	}
+	next();
+};
+
+@Service()
+@Authorized()
+@RestController('/dynamic-node-parameters')
+export class DynamicNodeParametersController {
+	constructor(private readonly service: DynamicNodeParametersService) {}
+
+	@Middleware()
```

Comment:
Not to change now, just to consider - it might be more consistent to have `@RestController('/dynamic-node-parameters', { middlewares: [fn] })` for controller-level middlewares, similar to endpoint-level middlewares.